packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -1,6 +1,46 @@
 import { Event } from './event';
+import { Expression } from './expression';
 import { CfnDetectorModel } from './iotevents.generated';
 
+/**
+ * Properties for options of state transition.
+ */
+export interface TransitionOptions {
+  /**
+   * The name of the event.
+   *
+   * @default string combining the names of the States as `${originStateName}_to_${targetStateName}`
+   */
+  readonly eventName?: string;
+
+  /**
+   * The condition that is used to determine to cause the state transition and the actions.
+   * When this was evaluated to TRUE, the state transition and the actions are triggered.
+   */
+  readonly when: Expression;
+}
+
+/**
+ * Specifies the state transition and the actions to be performed when the condition evaluates to TRUE.
+ */
+interface TransitionEvent {
+  /**
+   * The name of the event.
+   */
+  readonly eventName: string;
+
+  /**
+   * The condition that is used to determine to cause the state transition and the actions.
+   * When this was evaluated to TRUE, the state transition and the actions are triggered.
+   */
+  readonly when: Expression;
```

Comment:
It's fine to leave this name as `condition` (this is a package-private interface, it doesn't really matter what this name is - we can change it at any time. For that same reason, I'm pretty sure you can remove all of the documentation from this interface if you don't need it).